src/Microsoft.Data.Sqlite.Core/SqliteDataRecord.cs

```diff
@@ -170,9 +171,20 @@ public static Type GetFieldType(string type)
         }
 
         public virtual long GetBytes(int ordinal, long dataOffset, byte[] buffer, int bufferOffset, int length)
-            => throw new NotSupportedException();
+        {
+            var blob = GetBlob(ordinal);
+            Array.Copy(blob, dataOffset, buffer, bufferOffset, length);
+            return length;
+        }
 
         public virtual long GetChars(int ordinal, long dataOffset, char[] buffer, int bufferOffset, int length)
-            => throw new NotSupportedException();
+        {
+            var text = GetString(ordinal);
+            text.CopyTo((int)dataOffset, buffer, bufferOffset, length);
+            return length;
+        }
+
+        public virtual Stream GetStream(int ordinal)
+            => new MemoryStream(GetBlob(ordinal));
```

Comment:
I wonder if multiple calls should share the same buffer instance.

Reply:
Done.